### PR TITLE
Bug fix - Reference is no longer required (guidance)

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

Currently, the message saying a reference is ‘no longer required’ is shown to all referees 😱 

## Changes proposed in this pull request

- Update the controller so that the guidance is only shown when the reference has been cancelled.
- Add additional test coverage
- Address (silence?) Bullet gem ‘eager loading’ warning

## Guidance to review

- Please see bug descriptions in the trello ticket and confirm that they are now resolved. 🤞 

## Link to Trello card

https://trello.com/c/EHiwLtjp/2593-%F0%9F%92%942%EF%B8%8F%E2%83%A3-dev-improve-the-referees-journey-when-their-references-arent-required

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by george-bit — APPROVED** [review]
Looks sound 👍

File: app/services/cancel_referee.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class CancelReferee
   def call(reference:)
-    reference.update!(feedback_status: 'cancelled')
+    reference.reload.update!(feedback_status: 'cancelled')
```
Comment (davidgisbey) [inline app/services/cancel_referee.rb:3]: Not sure if this feels a bit wrong? Bullet being bullet i guess

Comment (TobyRet) [reply app/services/cancel_referee.rb:3]: Yeah I think you're right. 
The Bullet gem is raising an `eager loading` error on the spec when it comes to cancelling a reference. Both @malcolmbaig  and I have tried to investigate the cause of this to no avail -  we think it's possibly a false positive. Reloading makes the Bullet error disappear but then probably not so good we are modifying source code in this way for the sake of a test. The alternative is to disable bullet in the spec concerned and add a comment detailing findings from our investigation. What do you think?

Comment (davidgisbey) [reply app/services/cancel_referee.rb:3]: Yea I had to something similar in this spec candidate_interface/offers_and_withdrawls/candidate_withdraws_spec.rb a while ago.

image

I think not touching source code is preferable.

**Review by paulrobertlloyd — APPROVED** [review]

File: app/services/cancel_referee.rb (hunk 1 above)
Comment (TobyRet) [reply app/services/cancel_referee.rb]: Cool - I've disabled Bullet in the spec

**Pull request merged**